src/workers/personalizedDigestEmail.ts

```diff
@@ -129,10 +132,62 @@ const digestTypeToFunctionMap: Record<
   },
   [UserPersonalizedDigestType.ReadingReminder]: async (data, con) => {
     const { personalizedDigest, emailSendTimestamp, deduplicate = true } = data;
-    const emailSendDate = new Date(emailSendTimestamp);
+    const notificationSendTimestamp = new Date(emailSendTimestamp);
+    const currentDate = new Date();
+    await dedupedSend(
+      () =>
+        sendReadingReminderPush(
+          [personalizedDigest.userId],
+          notificationSendTimestamp,
+        ),
+      {
+        con,
+        personalizedDigest,
+        date: currentDate,
+        deduplicate,
+      },
+    );
+  },
+  [UserPersonalizedDigestType.StreakReminder]: async (data, con, logger) => {
+    const { personalizedDigest, emailSendTimestamp, deduplicate = true } = data;
+    const { userId } = personalizedDigest;
+
+    const notificationSendTimestamp = new Date(emailSendTimestamp);
     const currentDate = new Date();
+    const userSettings = await con
+      .getRepository(Settings)
+      .findOneBy({ userId });
+
+    // Safety measure to prevent sending streak reminders to users who have opted out
+    // but for some reason still have a streak reminder scheduled
+    if (userSettings?.optOutReadingStreak) {
+      return;
+    }
+
+    const userStreak = await con
+      .getRepository(UserStreak)
+      .findOneBy({ userId });
+
+    if (!userStreak) {
+      logger.error(
+        `User streak not found for user ${personalizedDigest.userId} when sending streak reminder.`,
+      );
```

Comment:
This is not really an error, maybe info, but once this runs in prod for few days I would just move it to debug level.

Reply:
Just changed it to debug